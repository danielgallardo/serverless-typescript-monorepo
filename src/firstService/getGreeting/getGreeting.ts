type Params = {
  name: string;
}

export const getGreeting = async ({name}: Params) => {
  return {
    message: `Hello ${name}! Welcome to Serverless`
  }
};
