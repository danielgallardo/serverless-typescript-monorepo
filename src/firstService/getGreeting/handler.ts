import middy from 'middy';
import {NormalizedEvent} from '../../@types';
import {apiRequestRoutine} from '../../common/middlewares/apiRequestRoutine';
import {eventValidator} from "../../common/middlewares/eventValidator";
import {logRoutine} from '../../common/middlewares/logRoutine';
import {Joi} from '../../common/validation';
import {getGreeting} from './getGreeting';

interface Event extends NormalizedEvent {
  queryStringParameters: {
    name: string;
  };
}

const schema = {
  queryStringParameters: Joi.object().keys({
    name: Joi.param()
  })
};

const handler = (event: Event) => {
  return getGreeting({
    name: event.queryStringParameters.name
  });
};

export default middy(handler)
  .use(apiRequestRoutine())
  .use(eventValidator(schema))
  .use(logRoutine());
