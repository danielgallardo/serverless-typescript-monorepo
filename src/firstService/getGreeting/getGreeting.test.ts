import {getGreeting} from './getGreeting';

describe('getGreeting', () => {
  it('should return correct greeting', () => {
    return expect(getGreeting({name: 'John'})).resolves.toMatchSnapshot();
  });
});
