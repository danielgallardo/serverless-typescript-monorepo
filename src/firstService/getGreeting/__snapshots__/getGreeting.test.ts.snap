// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getGreeting should return correct greeting 1`] = `
Object {
  "message": "Hello John! Welcome to Serverless",
}
`;
